ambry-api/src/main/java/com.github.ambry/router/GetBlobRange.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.router;
+
+/**
+ * Represents a byte range for performing ranged get requests.
+ */
+public class GetBlobRange {
+  public static final long UNDEFINED_OFFSET = -1;
+  private final long startOffset;
+  private final long endOffset;
+
+  /**
+   * Construct a range from the offsets parsed from the range header.
+   * @param startOffset the (inclusive) start byte offset, or {@link GetBlobRange#UNDEFINED_OFFSET}.
+   * @param endOffset the (inclusive) end byte offset, or {@link GetBlobRange#UNDEFINED_OFFSET}.
+   */
+  public GetBlobRange(long startOffset, long endOffset) {
```

Comment:
I don't think BlobRange should be a loose class like this. It should be coupled to a Blob and should do validation on it. If not, we should simply call this OffsetRange and then let BlobProperties do the validation for you based on size. You could add isValidOffsetRange to BlobProperties.